R/explain.R

```diff
@@ -251,15 +238,40 @@ explain.default <- function(model, data = NULL, y = NULL, predict_function = NUL
     }
   }
 
+  if (is.null(model_info)) {
+    # extract defaults
+    task_subtype <- check_if_multilabel(model, predict_function, data[1:2,])
+    model_info <- model_info(model, is_multiclass = task_subtype)
+    verbose_cat("  -> model_info        :  package", model_info$package[1], ", ver.", model_info$ver[1], ", task", model_info$type, "(", color_codes$yellow_start,"default",color_codes$yellow_end, ")", "\n", verbose = verbose)
+  } else {
+    verbose_cat("  -> model_info        :  package", model_info$package[1], ", ver.", model_info$ver[1], ", task", model_info$type, "\n", verbose = verbose)
+  }
+  # if type specified then it overwrite the type in model_info
+  if (!is.null(type)) {
+    model_info$type <- type
+    verbose_cat("  -> model_info        :  type set to ", type, "\n", verbose = verbose)
+  }
+
   # REPORT: checks for residual_function
   if (is.null(residual_function)) {
     # residual_function not specified
     # try the default
-    if (!is.null(predict_function)) {
+    if (!is.null(predict_function) & model_info$type != "multiclass") {
       residual_function <- function(model, data, y) {
         y - predict_function(model, data)
       }
       verbose_cat("  -> residual function :  difference between y and yhat (",color_codes$yellow_start,"default",color_codes$yellow_end,")\n", verbose = verbose)
+    } else if (!is.null(predict_function) & model_info$type == "multiclass") {
+      residual_function <- function(model, data, y) {
+        y_char <- as.character(y)
+        pred <- predict_function(model, data)
+        res <- c()
```

Comment:
`res` shall be initialized as a vector, like `numeric(length)` - it is much faster